[Ian Cheah] iP

IanCheah opened the pull request:
### Hyperion

> "Your mind is for having ideas, not holding them." - David Allen [source](https://medium.com/the-mission/the-most-popular-productivity-pieces-of-wisdom-from-david-allen-72ffe70ac7b9#:~:text=%E2%80%9CYour%20mind%20is%20for%20having%20ideas,%20not%20holding)

Hyperion frees your mind of having to remember things you need to do. This is your solution to a more productive day. It's

- text-based
- easy to learn
- **SUPER FAST** to use

All you need to do is

1. download it from here
2. double-click it.
3. add your tasks
4. seat back, and let it manage tasks for you 💯 

Most importantly, it is completely **FREE**!!!

Here is what we have to offer:

- [x] Managing tasks
- [x]  Managing deadlines
- [ ] Reminders (coming soon)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

IanCheah committed:
renamed Duke to Hyperion

IanCheah committed:
Greet and exits

IanCheah committed:
exits only when user types the command "bye"

IanCheah committed:
used a do while loop to validate user input

IanCheah committed:
included add and list logic

IanCheah committed:
implemented Task class

IanCheah committed:
functionality of mark and unmark event

IanCheah committed:
validation of user input

IanCheah committed:
created subclasses of task

IanCheah committed:
included support for todos, events, deadlines

IanCheah committed:
created automated testing

IanCheah committed:
created exception classes

IanCheah committed:
handled invalid input by throwing exceptions

IanCheah committed:
changed the array to arrayList to store tasks

IanCheah committed:
implemented delete tasks function

IanCheah committed:
added enum class

IanCheah committed:
created Bye class to handle bye logic

IanCheah committed:
created ListAll class to handle list logic

IanCheah committed:
edited ToDos class to handle todo logic

IanCheah committed:
edited Events class to handle event logic

IanCheah committed:
edited Deadlines class to handle deadline logic

IanCheah committed:
edited Bye class to handle bye logic

IanCheah committed:
created Delete class to handle delete logic

IanCheah committed:
created and exception to handle for no input at all

IanCheah committed:
created Mark class to handle mark logic

IanCheah committed:
created Unmark class to handle unmark logic

IanCheah committed:
created Parser class to handle for parsing logic

IanCheah committed:
edited Hyperion class for redeability

IanCheah committed:
edited expected input

IanCheah committed:
Add FileUtil.java

IanCheah committed:
Add loadTasks() and saveTasks() methods in Hyperion class

IanCheah committed:
Update Task.java class

IanCheah committed:
Merge branch 'branch-Level-7'

IanCheah committed:
Add time parsing

IanCheah committed:
Add time parsing for Event class

IanCheah committed:
Add inital printing of list of tasks

IanCheah committed:
Merge branch 'Level-8'

IanCheah committed:
Add ui class

IanCheah committed:
Add storage class

IanCheah committed:
update ToDos class

IanCheah committed:
update Deadlines class

IanCheah committed:
update Events class

IanCheah committed:
update Task class

IanCheah committed:
Update Hyperion class

IanCheah committed:
update parser class

IanCheah committed:
Add tasklists class

IanCheah committed:
delete FileUtil class

IanCheah committed:
Update Ui class to handle more functions

IanCheah committed:
Update the constructor of ToDos class

IanCheah committed:
Update Task class
removed implemntation of serializable

IanCheah committed:
Update parser class

IanCheah committed:
Hyperion class: replace type of allTasks to a TaskList

IanCheah committed:
Merge branch 'branch-A-MoreOOP'

IanCheah committed:
Delete Mark.java class

IanCheah committed:
Delete Unmark.java class

IanCheah committed:
Shift all files into packages

IanCheah committed:
Create exception package

IanCheah committed:
Create main package

IanCheah committed:
Create tasks package

IanCheah committed:
Merge remote-tracking branch 'origin/add-gradle-support'

IanCheah committed:
Remove all files under hyperion directory

IanCheah committed:
Update file path of main class in build.gradle

IanCheah committed:
Remove all files under hyperion directory

IanCheah committed:
Shift files from hyperion directory to java directory

IanCheah committed:
Include test for exception for Events class

IanCheah committed:
Update build.gradle to create hyperion.jar file

IanCheah committed:
Update EventsTest class

IanCheah committed:
Add javadoc for Parser class

IanCheah committed:
Add javadoc for ListAll class

IanCheah committed:
Add javadocs for Commands class

IanCheah committed:
Add javadoc for Storage class

IanCheah committed:
Add javadocs for Ui class

IanCheah committed:
Add javadocs for the various Task class

IanCheah committed:
Merge branch 'A-JavaDoc'

IanCheah committed:
Rename carryOn to isOver to comply with coding standard

IanCheah committed:
Rename carryOn to isOver to comply with coding standards

IanCheah committed:
Remove extra backslash

IanCheah committed:
Update javadoc

IanCheah committed:
Update javadoc of Parser.java

IanCheah committed:
Add find logic

IanCheah committed:
Merge branch 'Level-9'

IanCheah committed:
Remove unnecessary ListAll class

IanCheah committed:
Update System.out.println to Ui.display

IanCheah committed:
Add additional methods to TaskList

IanCheah committed:
Add display method and findMessage methods

IanCheah committed:
Update Storage class put method

IanCheah committed:
Merge branch 'branch-more-oop'

benjiBase commented on src/main/java/main/Hyperion.java:
```diff
@@ -0,0 +1,48 @@
+        Scanner scanner = new Scanner(System.in);
+        String input;
+
+        while(!isOver) {
```
while needs a space
while (!isOver)

IanCheah replied:
I had made the changes. Thanks!

benjiBase commented on src/main/java/task/ToDos.java:
```diff
@@ -0,0 +1,46 @@
+
+import java.util.List;
+
+public class ToDos extends Task{
```
need a spacebar after Task

IanCheah replied:
I had changed it. Thanks for spotting it for me!

benjiBase reviewed (commented):
LGTM. nice work

IanCheah committed:
Add checkstyle and suppression files

IanCheah committed:
Include checkstyle plugin in build.gradle

IanCheah committed:
Update checkstyle file

IanCheah committed:
Edit style issues for Parser class

IanCheah committed:
Edit the style of Storage class

IanCheah committed:
Edit style of TaskList according to checkstyle

IanCheah committed:
Update mark method's parameters

IanCheah committed:
Edit style for Hyperion class according to checkstyle

IanCheah committed:
Fix style issues in Parser class raised by stylecheck

IanCheah committed:
Fix style issues in Storage class raised by stylecheck

IanCheah committed:
Fix style issues in Deadlines class raised by stylecheck

IanCheah committed:
Fix style issues in Deadlines class specified by checkstyle

IanCheah committed:
Fix style issues in Events class raised by checkstyle

IanCheah committed:
Fix style issues in Ui class raised by checkstyle

IanCheah committed:
Fix style issues in TaskList class raised by checkstyle

IanCheah committed:
Fix style issues in ToDos class raised by checkstyle

IanCheah committed:
Fix style issues in Task class raised by checkstyle

IanCheah committed:
Fix style issues in CommandFoundButInvalidException class raised by checkstyle

IanCheah committed:
Fixed style issues in the various Exception class raised by checkstyle

IanCheah committed:
Fix style issues in Commands class raised by checkstyle

IanCheah committed:
Fix style issues in Test classes raised by checkstyle

IanCheah committed:
Merge branch 'branch-A-CodingStandard'

IanCheah committed:
Add new files for new gui

IanCheah committed:
Update starting point of the file

IanCheah committed:
Update files in gui package

IanCheah committed:
Edit files in the main package

IanCheah committed:
Update design of parser class by adding a run method

IanCheah committed:
Merge branch 'A-Level-10'

IanCheah committed:
Fix style issues raised by stylecheck

IanCheah committed:
Update list logic

IanCheah committed:
Add HyperionException class

IanCheah committed:
Update all Exception class to extend HyperionException

IanCheah committed:
Update Hyperion to catch HyperionException

riccoljy commented on src/main/java/main/Parser.java:
```diff
@@ -0,0 +1,104 @@
+    private String description;
+    private Storage storage;
+    private boolean isOver = false;
+    private Ui ui;
```
Extremely neat declaration of class attributes; nice!

riccoljy commented on src/main/java/main/Parser.java:
```diff
@@ -0,0 +1,104 @@
+            this.isOver = true;
+            return ui.bye();
+        default:
+            throw new CommandNotFoundException(command);
```
Very neat switch-case/default branches! Especially with how there's no fallthrough from `return`!

riccoljy commented on src/main/java/main/Storage.java:
```diff
@@ -0,0 +1,105 @@
+                    currTask = new Events(parts[2]);
+                    break;
+                default:
+                    throw new InvalidSyntaxException("File is corrupted");
```
Nice touch accounting for corrupted save file

riccoljy commented on src/main/java/main/Storage.java:
```diff
@@ -0,0 +1,105 @@
+                default:
+                    throw new InvalidSyntaxException("File is corrupted");
+                }
+                if (Integer.parseInt(parts[1].trim()) == 1) {
```
`parts[1].trim()` seems a bit unclear what it does. Perhaps another way to save whether the task is done, like `"T"`/`"F"` or `"Done"`/'`Undone"`? Perhaps another method to parse it to show whether it is done or not?

riccoljy commented on src/main/java/main/TaskList.java:
```diff
@@ -0,0 +1,236 @@
+    private Stack<Task> deletedTasks;
+    private Stack<Task> addedTasks;
+    private Stack<Task> markedTasks;
+    private Stack<Task> unmarkedTasks;
```
Very neat declaration once again, nice!

riccoljy commented on src/main/java/main/Ui.java:
```diff
@@ -0,0 +1,99 @@
+    public String addedMessage(Task t, int size) {
+        String str1 = "Got it. I've added this task:\n";
+        String str2 = String.format("Now you have %d tasks in the list", size);
+        return str1 + t.toString() + "\n" + str2;
```
Minor nit but `.toString()` isn't necessary

riccoljy reviewed (commented):
Very well written and neat. Nice!

IanCheah committed:
Update welcome message

IanCheah committed:
Include assertions in methods contains in TaskList class

The unmark, mark method expects an index that is in range.
The list method expects no input after the command.

We include assertions here to ensure that in the main logic of the
methods, the inputs are valid.

IanCheah committed:
Remove display(String message) method in Ui class

The strings are returned from the methods contained in the Ui class.
There is no need for a display method.

Removing the redundant display method improves code quality.

IanCheah committed:
Remove isOver() method within Parser class.

There is no need to close the program if the user types bye.
Users are able to exit the program via the exit button.

Removing the redundant isOver() method improves code quality.

IanCheah committed:
Remove obvious comments in the Deadlines class

Comments like "check if is a valid LocalDate" is obvious from the
code.

Removing such a comment improves code quality.

IanCheah committed:
Remove unnecessaryary exception frthe om constructor

The exception was no longer required as the constructor's logic was
updated

IanCheah committed:
Use stream to imporve find(String str) method in TaskList

Replaced the loop-based filtering with a stream API to make the
code clearer and more efficient

IanCheah committed:
Merge pull request #2 from IanCheah/branch-A-Assertions

Include assertions in methods contains in TaskList class

IanCheah committed:
Merge pull request #3 from IanCheah/branch-A-CodeQuality

Improved code quality

IanCheah committed:
Merge pull request #4 from IanCheah/branch-A-Streams

Use stream to imporve find(String str) method in TaskList

IanCheah committed:
Merge branch 'master' of https://github.com/IanCheah/ip

* 'master' of https://github.com/IanCheah/ip:
  Use stream to imporve find(String str) method in TaskList
  Remove unnecessaryary exception frthe om constructor
  Remove obvious comments in the Deadlines class
  Remove isOver() method within Parser class.
  Remove display(String message) method in Ui class
  Include assertions in methods contains in TaskList class

IanCheah committed:
Add gradle.yml file to set up continuous integration

IanCheah committed:
Add relavant information in gradle.yml file

IanCheah committed:
Add TaskComparator to support task sorting by type and criteria

Implemented TaskComparator to sort tasks with the following order:
1. ToDos sorted by description in ASCII order
2. Deadlines sorted by ascending due date
3. Events sorted by ascending start date

IanCheah committed:
Update files to handle sorting logic

IanCheah committed:
Merge branch 'branch-BCD-Extension'

IanCheah committed:
Add multiple Command classes

Currently the run() method in the parser class does not adhere to
the SLAP principle. There are various levels of abstractions in the
run() method.

Multiple Command classes that inherits the Command interface are
added so that the logic are implemented in their respective Command
class. This improves code quality.

IanCheah committed:
Add implementation to the various command classes

IanCheah committed:
Merge branch 'branch-Even-More-OOP'

IanCheah committed:
Remove DaUser and Hyperion picture

IanCheah committed:
Update DaUser and Hyperion picture

IanCheah committed:
Update DaUser and Hyperion picture

IanCheah committed:
Add css stylesheet for dialog-box and main

IanCheah committed:
Include details for styling in the css files for dialog-box and
main

IanCheah committed:
Add a new background for chatbot

IanCheah committed:
Merge branch 'branch-A-BetterGui'

IanCheah committed:
Add personality for the chatbot

IanCheah committed:
Merge branch 'A-Personality'

IanCheah committed:
Add extra junit test for the event class

IanCheah committed:
Merge branch 'branch-A-MoreTesting'

IanCheah committed:
Fix style errors that were raised by checkstyle

IanCheah committed:
Edit name of boolean variable

IanCheah committed:
Fix minor style erros

IanCheah committed:
Add helper function to reduce SLAP violation in load() method of
storage class

IanCheah committed:
Add error handling for the case where start date is after end date
for event

IanCheah committed:
Merge branch 'branch-A-MoreErrorHandling'

IanCheah committed:
Add an image that is used in README.md

IanCheah committed:
Update README.md for Hyperion user guide

IanCheah committed:
Fix minor bug

IanCheah committed:
Rename image of screenshot to Ui.png

IanCheah committed:
Add title to GUI

IanCheah committed:
Update Ui.png

IanCheah committed:
Add bye functionality where the program closes 2 seconds after
the bye command is sent

IanCheah committed:
Update README.md file

IanCheah committed:
Remove unecessary constructor
